Update AnnotTrack.js

gw161 opened the pull request:
Added methods to utilize API featuresByName. However isn't working as intended.

Commit:
Update AnnotTrack.js

Added methods to utilize API featuresByName. However isn't working as intended.

Commit:
Update AnnotTrack.js

nathandunn commented:
I don't think you need to talk to the server at all. 

Something like below:

```
var seqtrack = track.getSequenceTrack();
seqtrack.sequenceStore.getReferenceSequence(  { ref: track.refSeq.name, start: feature.fmin - 1000, end: feature.fmax + 1000 }, function (seq){
     //  some function like in SequenceTrack, but for your use-case
   }
```

Commit:
Update AnnotTrack.js

Actually figured that the records contains information about exon positions. So built a couple lines to parse it from there and then an un-optimized for loop structure to generate some numbering.

gw161 commented:
@nathandunn 
Actually made some progress today on the numbering. It is all contained within an array for the moment. Perhaps it should be placed into the records json object?

You see anything totally egregious with this? 

Also need to figure out how to start numbering from the first A in the ATG. This information is usually held in genbank records. 

I was thinking maybe the Annotation Track onmouseover could be modified from not only showing the owner but also this numbering and numbering of genomic position as well.

Commit:
Update AnnotTrack.js

nathandunn commented:
> On Nov 9, 2017, at 11:10 AM, Wozzerv <notifications@github.com> wrote:
> 
> @nathandunn <https://github.com/nathandunn>
> Actually made some progress today on the numbering. It is all contained within an array for the moment. Perhaps it should be placed into the records json object?
> 
If it seems cleaner, sure.  
> You see anything totally egregious with this?
> 
Not sure why you have a “this.this” call. 

Also, I would prefer (but its your code) to use es6 features for each vs jQuery, but that is not particularly :

https://hacks.mozilla.org/2015/04/es6-in-depth-iterators-and-the-for-of-loop/ <https://hacks.mozilla.org/2015/04/es6-in-depth-iterators-and-the-for-of-loop/>

Really, everything seems to look reasonable, though.
> Also need to figure out how to start numbering from the first A in the ATG. This information is usually held in genbank records.
> 
> I was thinking maybe the Annotation Track onmouseover could be modified from not only showing the owner but also this numbering and numbering of genomic position as well.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/Wozzerv/Apollo/pull/1#issuecomment-343222356>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAt2qjOpspstRvtCzWGtMA73JMeJaDJRks5s0zDrgaJpZM4QW6Pf>.
>
